### PR TITLE
add igc external package

### PR DESCRIPTION
**Pull request by rscohn2**
Enabling linking with igc through a package that relies on an external installation.

@jmellorcrummey: Please try this.

You need to add a:
```
depends_on('igc')
```

and add this to your ``~/.spack/packages.yaml``
```
packages:
  oneapi-igc-external:
    externals:
    - spec: oneapi-igc-external@1.0.10409
      prefix: /usr
```

### CODE REVIEW

**Comment by rscohn2** [comment]
@jmellorcrummey : I changed the package name to oneapi-igc, and got rid of the igc virtual.

You need to add a:
```
depends_on('oneapi-igc')
```

and add this to your ``~/.spack/packages.yaml``
```
packages:
  oneapi-igc:
    externals:
    - spec: oneapi-igc@1.0.10409
      prefix: /usr
```

**Comment by jmellorcrummey** [comment]
@rscohn2 this is on my list of things to test with my colleague @mwkrentel. We want to test this in conjunction with the level0, gtpin, and opencl headers packages. we should be able to get to do some testing in the next few days.

**Comment by jmellorcrummey** [comment]
@rscohn2 

TL;DR: this package works as intended and meets our needs. i see us being ready to push a spack recipe that depends upon this oneapi-igc package next week.

i have a bleeding-edge development branch of hpctoolkit that we successfully built with this oneapi-igc package. we used to collect measurement data to verify that everything was proper. however, i am not yet ready to push my bleeding-edge development branch and its spack recipe. we have to do some further testing before merging it into our shared "develop" branch. when we do that. we will push a spack recipe that depends upon oneapi-igc.

**Comment by rscohn2** [comment]
@spackbot run pipeline

**Comment by rscohn2** [comment]
> TL;DR: this package works as intended and meets our needs. i see us being ready to push a spack recipe that depends upon this oneapi-igc package next week.

Thanks for your testing, persistence, and patience. It is the solution that @mwkrentel proposed a long time ago.

**Review by scheibelp — APPROVED** [review]
